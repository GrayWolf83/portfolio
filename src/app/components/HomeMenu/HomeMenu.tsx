import React from 'react'
import {
	Typography,
	Grid,
	Card,
	Avatar,
	CardContent,
	CardActions,
	Button,
	Link as MUILink,
} from '@mui/material'
import Search from '@mui/icons-material/Search'
import { Link } from 'react-router-dom'
import { Email, Telegram } from '@mui/icons-material'

interface IHomeMenu {}

const HomeMenu: React.FC<IHomeMenu> = () => {
	return (
		<Grid container spacing={2} data-testid='HomeMenu'>
			<Grid item xs={12} md={4}>
				<Card sx={{ height: '100%' }}>
					<Avatar
						src='https://avatars.githubusercontent.com/u/78337844?v=4'
						sx={{
							width: '150px',
							height: '150px',
							m: '10px auto',
						}}
					/>
					<CardContent>
						<Typography
							gutterBottom
							color='primary'
							variant='h6'
							component='div'>
							Мое портфолио
						</Typography>
					</CardContent>
					<CardActions>
						<Link to='/user/GrayWolf83'>
							<Button size='small' variant='contained'>
								Открыть
							</Button>
						</Link>
					</CardActions>
				</Card>
			</Grid>
			<Grid item xs={12} md={4}>
				<Card sx={{ height: '100%' }}>
					<Search
						sx={{
							fontSize: '150px',
							m: '10px auto',
							display: 'block',
						}}
						color='primary'
					/>
					<CardContent>
						<Typography
							gutterBottom
							color='primary'
							variant='h6'
							component='div'>
							Поиск пользователей
						</Typography>
					</CardContent>
					<CardActions>
						<Link to='/search'>
							<Button size='small' variant='contained'>
								Открыть
							</Button>
						</Link>
					</CardActions>
				</Card>
			</Grid>
			<Grid item xs={12} md={4}>
				<Card sx={{ height: '100%' }}>
					<CardContent>
						<Typography
							gutterBottom
							color='primary'
							variant='h6'
							component='div'>
							Мои контакты
						</Typography>
						<MUILink
							href='mailto: sb_aktau@bk.ru'
							sx={{ display: 'flex', mb: '10px' }}>
							<Email />
							<Typography sx={{ ml: '5px' }}>Email</Typography>
						</MUILink>
						<MUILink
							href='https://tlgg.ru/@SergeyBernyakovich'
							target='_blank'
							sx={{ display: 'flex' }}>
							<Telegram />
							<Typography sx={{ ml: '5px' }}>
								Telegramm
							</Typography>
						</MUILink>
					</CardContent>
				</Card>
			</Grid>
		</Grid>
	)
}

export default HomeMenu
