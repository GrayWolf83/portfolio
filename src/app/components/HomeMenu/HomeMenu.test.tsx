import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import HomeMenu from './HomeMenu'

describe('HomeMenu component', () => {
	test('HomeMenu render', () => {
		render(
			<BrowserRouter>
				<HomeMenu />
			</BrowserRouter>,
		)

		const homeMenu = screen.getByTestId('HomeMenu')
		expect(homeMenu).toBeInTheDocument()
	})

	test('HomeMenu snapshot', () => {
		expect(
			render(
				<BrowserRouter>
					<HomeMenu />
				</BrowserRouter>,
			),
		).toMatchSnapshot()
	})
})
