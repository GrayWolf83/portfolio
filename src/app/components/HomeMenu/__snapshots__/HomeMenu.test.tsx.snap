// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeMenu component HomeMenu snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        data-testid="HomeMenu"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-b0s0xk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://avatars.githubusercontent.com/u/78337844?v=4"
              />
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
              >
                Мое портфолио
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <a
                href="/user/GrayWolf83"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Открыть
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-3tvku1-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
              >
                Поиск пользователей
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <a
                href="/search"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Открыть
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
              >
                Мои контакты
              </div>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-15z76vi-MuiTypography-root-MuiLink-root"
                href="mailto: sb_aktau@bk.ru"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="EmailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1ybbsdq-MuiTypography-root"
                >
                  Email
                </p>
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1dkk3ib-MuiTypography-root-MuiLink-root"
                href="https://tlgg.ru/@SergeyBernyakovich"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TelegramIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9.78 18.65l.28-4.23 7.68-6.92c.34-.31-.07-.46-.52-.19L7.74 13.3 3.64 12c-.88-.25-.89-.86.2-1.3l15.97-6.16c.73-.33 1.43.18 1.15 1.3l-2.72 12.81c-.19.91-.74 1.13-1.5.71L12.6 16.3l-1.99 1.93c-.23.23-.42.42-.83.42z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1ybbsdq-MuiTypography-root"
                >
                  Telegramm
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      data-testid="HomeMenu"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-b0s0xk-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://avatars.githubusercontent.com/u/78337844?v=4"
            />
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
            >
              Мое портфолио
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <a
              href="/user/GrayWolf83"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Открыть
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-3tvku1-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
            >
              Поиск пользователей
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <a
              href="/search"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Открыть
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
            >
              Мои контакты
            </div>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-15z76vi-MuiTypography-root-MuiLink-root"
              href="mailto: sb_aktau@bk.ru"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ybbsdq-MuiTypography-root"
              >
                Email
              </p>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1dkk3ib-MuiTypography-root-MuiLink-root"
              href="https://tlgg.ru/@SergeyBernyakovich"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="TelegramIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.78 18.65l.28-4.23 7.68-6.92c.34-.31-.07-.46-.52-.19L7.74 13.3 3.64 12c-.88-.25-.89-.86.2-1.3l15.97-6.16c.73-.33 1.43.18 1.15 1.3l-2.72 12.81c-.19.91-.74 1.13-1.5.71L12.6 16.3l-1.99 1.93c-.23.23-.42.42-.83.42z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ybbsdq-MuiTypography-root"
              >
                Telegramm
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
